specifyweb/frontend/js_src/lib/components/WorkBench/index.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useAsyncState } from '../../hooks/useAsyncState';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { useErrorContext } from '../../hooks/useErrorContext';
+import { commonText } from '../../localization/common';
+import { wbText } from '../../localization/workbench';
+import { ajax } from '../../utils/ajax';
+import { f } from '../../utils/functools';
+import type { GetSet } from '../../utils/types';
+import { LoadingContext } from '../Core/Contexts';
+import { useMenuItem } from '../Header/MenuContext';
+import { treeRanksPromise } from '../InitialContext/treeRanks';
+import { LoadingScreen } from '../Molecules/Dialog';
+import { Dialog } from '../Molecules/Dialog';
+import { NotFoundView } from '../Router/NotFoundView';
+import type { Dataset } from '../WbPlanView/Wrapped';
+import { WbView } from './WbView';
+
+export function WorkBench(): JSX.Element {
+  useMenuItem('workBench');
+
+  const [treeRanksLoaded = false] = useAsyncState(fetchTreeRanks, true);
+  const { id } = useParams();
+  const datasetId = f.parseInt(id);
+
+  const [dataset, setDataset] = useDataset(datasetId);
+  useErrorContext('dataSet', dataset);
+  const loading = React.useContext(LoadingContext);
+
+  const [isDeleted, handleDeleted] = useBooleanState();
+  // @ts-expect-error figure out how handleDeletedConfirmation was being used in Backbone. possibly not used at all
+  const [isDeletedConfirmation, handleDeletedConfirmation] = useBooleanState();
+
+  const navigate = useNavigate();
+
+  if (dataset === undefined || !treeRanksLoaded) return <LoadingScreen />;
+
+  const triggerDatasetRefresh = () => {
+    loading(fetchDataset(dataset.id).then(setDataset));
+  };
+
+  return datasetId === undefined ? (
+    <NotFoundView />
+  ) : isDeleted ? (
+    <>{wbText.dataSetDeletedOrNotFound()}</>
+  ) : isDeletedConfirmation ? (
+    <Dialog
+      buttons={commonText.close()}
+      header={wbText.dataSetDeleted()}
+      onClose={(): void => navigate('/specify/', { replace: true })}
+    >
+      {wbText.dataSetDeletedDescription()}
+    </Dialog>
```

Comment:
Is this ever rendered/when is this expected to be rendered? In the old WB, I couldn't find any place where `handleDeletedConfirmation()` is called and so I don't think this is needed but I'm not sure what the expected behavior is. Also, I think delete confirmation comes from `DataSetMeta` instead?

Reply:
yes, there should probably me a confirmation.
if there wasn't one before that might a bug

Reply:
https://github.com/specify/specify7/blob/4e576e710978a8bd1c96ce2f6091ebe5b236e8ef/specifyweb/frontend/js_src/lib/components/WorkBench/DataSetMeta.tsx#L128-L135

We have the same dialog box in `DataSetMeta` which is what's actually being used when deleting the dataset so I think we don't need this code here in `index.tsx`